From 94f8feff2f43c5a03a435297b8fba140d04deefe Mon Sep 17 00:00:00 2001
From: Pig <pig.priv@gmail.com>
Date: Thu, 15 Oct 2020 08:44:20 +0800
Subject: [PATCH 12/12] camera: Allow devices to load custom CameraParameter
 code

* Some devices need additional code to load their cameras. Add a hook
  for extra classes and symbols to be included.

Originally implemented in Android.mk at 38b27bf2e5079441358288402b38865ba7c1d9d6
Credit to Ethan Chen for original Android.mk implementation
Originally implemented via lineage product variable at 7398f91e051086bdd210a8bae0533f15fac1a37f
Credit to Rashed Abdel-Tawab for original product variable implementation

Change-Id: Iec667af2f0bfdb41f14df342cb128908244d7af0
---
 camera/Android.bp | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/camera/Android.bp b/camera/Android.bp
index 2b05a844bf..1924e80b78 100644
--- a/camera/Android.bp
+++ b/camera/Android.bp
@@ -45,6 +45,7 @@ license {
 
 cc_library_shared {
     name: "libcamera_client",
+    defaults: ["camera_parameter_library_defaults"],
 
     aidl: {
         export_aidl_headers: true,
@@ -63,7 +64,6 @@ cc_library_shared {
         // Source for camera interface parcelables, and manually-written interfaces
         "Camera.cpp",
         "CameraMetadata.cpp",
-        "CameraParameters.cpp",
         "CaptureResult.cpp",
         "CameraParameters2.cpp",
         "CameraSessionStats.cpp",
@@ -110,6 +110,16 @@ cc_library_shared {
 
 }
 
+cc_library_static {
+    name: "libcamera_parameters",
+
+    export_include_dirs: [
+        "include",
+    ],
+    srcs: ["CameraParameters.cpp"],
+
+}
+
 // AIDL interface between camera clients and the camera service.
 filegroup {
     name: "libcamera_client_aidl",
-- 
2.33.1

